src/rdkafka_cgrp.c

```diff
@@ -2784,12 +2874,20 @@ static void rd_kafka_cgrp_offsets_commit (rd_kafka_cgrp_t *rkcg,
         rd_kafka_buf_t *rkbuf;
         rd_kafka_op_t *reply;
 
-        /* If offsets is NULL we shall use the current assignment. */
+        /* If offsets is NULL we shall use the current assignment
+         * (not the group assignment). */
         if (!rko->rko_u.offset_commit.partitions &&
-            rkcg->rkcg_assignment.all->cnt > 0)
-		rko->rko_u.offset_commit.partitions =
-			rd_kafka_topic_partition_list_copy(
-				rkcg->rkcg_assignment.all);
+            rkcg->rkcg_rk->rk_consumer.assignment.all->cnt > 0) {
+                if (rd_kafka_cgrp_assignment_is_lost(rkcg)) {
+                        /* Not committing assigned offsets: assignment lost */
+                        err = RD_KAFKA_RESP_ERR__ASSIGNMENT_LOST;
```

Comment:
is consumer poll ever going to spit this out (if auto committing?) - if yes (i didn't look), then the user has two ways to be notified of this - once by way of the rebalance_cb and once via a consumer error. I'd argue a consumer error isn't necessary.

tbh, this comment is also driven by a latent general uneasy feeling about having both error_cb and error messages returned from poll - both are delivered inline - having two mechanisms seems not great to me.

Reply:
This will be propagated through the offset_commit_cb (if async commit) or through commit()'s return value.